pkg/quantile/ddsketch.go

```diff
@@ -0,0 +1,206 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package quantile
+
+import (
+	"fmt"
+	"math"
+	"sort"
+
+	"github.com/DataDog/datadog-agent/pkg/quantile/summary"
+	"github.com/DataDog/sketches-go/ddsketch"
+	"github.com/DataDog/sketches-go/ddsketch/mapping"
+	"github.com/DataDog/sketches-go/ddsketch/store"
+)
+
+const (
+	maxIndex = math.MaxInt16
+)
+
+// convertToCompatibleDDSketch converts any DDSketch into a DDSketch with parameters
+// compatible with Sketch
+func convertToCompatibleDDSketch(c *Config, inputSketch *ddsketch.DDSketch) (*ddsketch.DDSketch, error) {
+	// Create positive store for the new DDSketch
+	positiveStore := store.NewDenseStore()
+
+	// Create negative store for the new DDSketch
+	negativeStore := store.NewDenseStore()
+
+	gamma := c.gamma.v
+	// TODO: using 1338 instead of 1338.5 gave better results for DDSketch -> Sketch conversion
+	//       Why? May have to do with the way we convert float counts to int counts in
+	//       keyCountsFromFloatKeyCounts.
+	offset := float64(c.norm.bias)
+	newMapping, err := mapping.NewLogarithmicMappingWithGamma(gamma, offset)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create LogarithmicMapping for DDSketch: %v", err)
```

Comment:
Use `%w` for a [wrapped error](https://pkg.go.dev/errors#pkg-overview)
```suggestion
		return nil, fmt.Errorf("couldn't create LogarithmicMapping for DDSketch: %w", err)
```